### PR TITLE
change types to bool in globals.h

### PR DESCRIPTION
**Pull request by cjoach**

### CODE REVIEW

File: src/nvim/globals.h [hunk 1]
```diff
@@ -1086,8 +1000,8 @@ EXTERN char psepcN INIT(= '/');            // abnormal path separator character
 EXTERN char pseps[2] INIT(= { '\\', 0 });  // normal path separator string
 #endif
 
-/* Set to TRUE when an operator is being executed with virtual editing, MAYBE
- * when no operator is being executed, FALSE otherwise. */
+// Set to bool when an operator is being executed with virtual editing, MAYBE
```
Comment (ZyX-I) [inline src/nvim/globals.h]: `TRUE` replaced with `bool`, not `true`.

Comment (bfredl) [inline src/nvim/globals.h]: Though a tri-state better not use `true`/`false`. Either use `TriState` or refactor to keep track of "no operator being executed" otherhow.

Comment (cacplate) [inline src/nvim/globals.h]: I missed the `TriState` on this one, will revert to int for the time being. I intend to change the multiple tristates in the file as well in another PR.

**Comment by justinmk** [comment]
Is this taking into account the discussion at https://github.com/neovim/neovim/pull/3917 ?

**Comment by bfredl** [comment]
The situation at #3917 is somewhat different as it concerns variables behind variant pointers (and we need to guard against an `int*` being accessed as a `bool*` or vice-versa, while assing `true` to a `TriState` is "just" a style error)

**Pull request closed without merging**